101
파이썬에서 True 혹은 False를 갖는 데이터 타입은 무엇인가?

boolean

102
아래 코드의 출력 결과를 예상하라

print(3 == 5)

In [2]:
print(3 == 5)

False


103
아래 코드의 출력 결과를 예상하라

print(3 < 5)


In [3]:
print(3 < 5)

True


104  
아래 코드의 결과를 예상하라.   

x = 4   
print(1 < x < 5)   

In [5]:
x = 4
print(1 < x < 5)

True


105
아래 코드의 결과를 예상하라.

print ((3 == 3) and (4 != 3))

In [6]:
print ((3 == 3) and (4 != 3))

True


106
아래 코드에서 에러가 발생하는 원인에 대해 설명하라.

print(3 => 4)

In [7]:
print(3 => 4)

SyntaxError: expression cannot contain assignment, perhaps you meant "=="? (3612698823.py, line 1)

">="라고 쓰던지 참..

107
아래 코드의 출력 결과를 예상하라

if 4 < 3:    
    print("Hello World")


In [8]:
if 4 < 3:
    print("Hello World")

조건을 만족하지 않아서 프린트문은 실행되지않음

108
아래 코드의 출력 결과를 예상하라    

if 4 < 3:   
    print("Hello World.")    
else:   
    print("Hi, there.")

In [9]:
if 4 < 3:
    print("Hello World.")
else:
    print("Hi, there.")


Hi, there.


109
아래 코드의 출력 결과를 예상하라    
   
if True :    
    print ("1")   
    print ("2")  
else :  
    print("3")  
print("4")   

In [10]:
if True :
    print ("1")
    print ("2")
else :
    print("3")
print("4")

1
2
4


110
아래 코드의 출력 결과를 예상하라   

if True :   
    if False:   
        print("1")   
        print("2")   
    else:   
        print("3")   
else :   
    print("4")   
print("5")   

In [11]:
if True :
    if False:
        print("1")
        print("2")
    else:
        print("3")
else :
    print("4")
print("5")

3
5


111
사용자로부터 입력받은 문자열을 두 번 출력하라. 아래는 사용자가 "안녕하세요"를 입력한 경우의 출력 결과이다.

In [15]:
user = input("입력:")
print(user * 2)

입력:안녕하세요
안녕하세요안녕하세요


112
사용자로부터 하나의 숫자를 입력받고, 입력 받은 숫자에 10을 더해 출력하라.

In [16]:
number=int(input())
print(number+10)

30
40


113
사용자로부터 하나의 숫자를 입력 받고 짝수/홀수를 판별하라.

In [18]:
is_odd=int(input())
if is_odd %2==0:
    print("짝수")
else:
    print("홀수")

30
짝수


114
사용자로부터 값을 입력받은 후 해당 값에 20을 더한 값을 출력하라.    
단 사용자가 입력한 값과 20을 더한 계산 값이 255를 초과하는 경우 255를 출력해야 한다.

In [21]:
num=input("입력값:")
plus20=int(num)+20

if plus20 >255:
    print(255)
else:
    print(plus20)

입력값:264
255


115
사용자로부터 하나의 값을 입력받은 후 해당 값에 20을 뺀 값을 출력하라.   
단 출력 값의 범위는 0~255이다. 예를 들어 결괏값이 0보다 작은 값이되는 경우 0을 출력하고 255보다 큰 값이 되는 경우 255를 출력해야 한다.

In [22]:
num=input("입력값:")
minus20=int(num)-20

if minus20<0:
    print(0)
elif minus20>255:
    print(255)
else:
    print(minus20)

입력값:15
0


116
사용자로부터 입력 받은 시간이 정각인지 판별하라.

>> 현재시간:02:00
정각 입니다.

In [25]:
time=input("현재시각:")
if time[3:]=="00":
    print("정각입니다")
else:
    print("정각이 아닙니다")

현재시각:16:38
정각이 아닙니다


117
사용자로 입력받은 단어가 아래 fruit 리스트에 포함되어 있는지를 확인하라. 포함되었다면 "정답입니다"를 아닐 경우 "오답입니다" 출력하라.

fruit = ["사과", "포도", "홍시"]

In [26]:
fruit = ["사과", "포도", "홍시"]
fav_fruit=input("종아하는 과일은?")

if fav_fruit in fruit:
    print("정답입니다")
else:
    print("오답입니다")

종아하는 과일은?거봉
오답입니다


118
투자 경고 종목 리스트가 있을 때 사용자로부터 종목명을 입력 받은 후 해당 종목이 투자 경고 종목이라면 '투자 경고 종목입니다'를 아니면 "투자 경고 종목이 아닙니다."를 출력하는 프로그램을 작성하라.

warn_investment_list = ["Microsoft", "Google", "Naver", "Kakao", "SAMSUNG", "LG"]

In [28]:
warn_investment_list = ["Microsoft", "Google", "Naver", "Kakao", "SAMSUNG", "LG"]
invest=input("투자종목:")

if invest in warn_investment_list:
    print("투자경고 종목입니다.")
else:
    print("투자경고 종목이 아닙니다")

투자종목:EstSoft
투자경고 종목이 아닙니다


119
아래와 같이 fruit 딕셔너리가 정의되어 있다. 사용자가 입력한 값이 딕셔너리 키 (key) 값에 포함되었다면 "정답입니다"를 아닐 경우 "오답입니다" 출력하라.

fruit = {"봄" : "딸기", "여름" : "토마토", "가을" : "사과"}

In [31]:
fruit = {"봄" : "딸기", "여름" : "토마토", "가을" : "사과"}
user = input("제가좋아하는계절은: ")
if user in fruit.keys():
    print("정답입니다.")
else:
    print("오답입니다.")

제가좋아하는계절은: 봄동
오답입니다.


120
아래와 같이 fruit 딕셔너리가 정의되어 있다. 사용자가 입력한 값이 딕셔너리 값 (value)에 포함되었다면 "정답입니다"를 아닐 경우 "오답입니다" 출력하라.

fruit = {"봄" : "딸기", "여름" : "토마토", "가을" : "사과"}

In [33]:
fruit = {"봄" : "딸기", "여름" : "토마토", "가을" : "사과"}
user = input("좋아하는 과일은?")
if user in fruit.values():
    print("정답입니다.")
else:
    print("오답입니다.")

좋아하는 과일은?도마도
오답입니다.


121
사용자로부터 문자 한 개를 입력 받고, 소문자일 경우 대문자로, 대문자 일 경우, 소문자로 변경해서 출력하라.

>> a
A   


힌트-1 : islower() 함수는 문자의 소문자 여부를 판별합니다. 만약 소문자일 경우 True, 대문자일 경우 False를 반환합니다.    
힌트-2 : upper() 함수는 대문자로, lower() 함수는 소문자로 변경합니다.

In [35]:
alphabet=input()
if alphabet.islower==True:
    print(alphabet.upper())
else:
    print(alphabet.lower())

A
a


122     
점수 구간에 해당하는 학점이 아래와 같이 정의되어 있다. 사용자로부터 score를 입력받아 학점을 출력하라.     

점수	학점        
81~100	A          
61~80	B       
41~60	C        
21~40	D       
0~20	E       

In [37]:
score=int(input())
if score>=81:
    print("grade is A")
elif 61<=score<=80:
    print("grade is B")
elif 41<=score<=60:
    print("grade is C")
elif 21<=score<=40:
    print("grade is D")
else:
    print("grade is E")

62
grade is B


123      
사용자로부터 달러, 엔, 유로, 또는 위안 금액을 입력받은 후 이를 원으로 변환하는 프로그램을 작성하라.     
각 통화별 환율은 다음과 같다. 사용자는 100 달러, 1000 엔, 13 유로, 100 위안과 같이 금액과 통화명 사이에 공백을 넣어 입력한다고 가정한다.

통화명	환율        
달러	1167         
엔	1.096        
유로	1268        
위안	171      

In [38]:
환율 = {"달러": 1167, 
        "엔": 1.096, 
        "유로": 1268, 
        "위안": 171}

In [46]:
money=input("입력: ")
num, currency = money.split() # 숫자와 단위 분할
print(float(num)* 환율[currency], "원")

입력: 123 엔
134.80800000000002 원


124
사용자로부터 세 개의 숫자를 입력 받은 후 가장 큰 숫자를 출력하라.

>> input number1: 10   
>> input number2: 9   
>> input number3: 20  
20


In [50]:
num1=int(input("num1:"))
num2=int(input("num2:"))
num3=int(input("num3:"))

if num1>=num2 and num1>=num3:
    print(num1)
elif num2>=num1 and num2>=num3:
    print(num2)
elif num3>=num1 and num3>=num2:
    print(num3)

num1:126
num2:124
num3:126
126


125
휴대폰 번호 앞자리에 따라 통신사는 아래와 같이 구분된다. 사용자로부터 휴대전화 번호를 입력 받고, 통신사를 출력하는 프로그램을 작성하라.

번호	통신사    
011	SKT   
016	KT   
019	LGU   
010	알수없음  


In [56]:
phone=str(input())
if phone[:3]=="011":
    print("SKT")
elif phone[:3]=="016":
    print("KT")
elif phone[:3]=="019":
    print("LG")
elif phone[:3]=="010":
    print("알수없음")

010-1234-5678
알수없음


126
우편번호는 5자리로 구성되는데, 앞의 세자리는 구를 나타낸다. 예를들어, 강북구의 경우 010, 011, 012 세 자리로 시작한다.

-	0	1	2	3	4	5	6	7	8	9
01	강북구	강북구	강북구	도봉구	도봉구	도봉구	노원구	노원구	노원구	노원구
사용자로 부터 5자리 우편번호를 입력받고 구를 판별하라

>> 우편번호: 01400
도봉구

In [63]:
우편번호 = input("우편번호: ")
우편번호 = 우편번호[:3]
if 우편번호 in ["010", "011", "012"]:
    print("강북구")
elif 우편번호 in ["014", "015", "016"]:
    print("도봉구")
else:
    print("노원구")

우편번호: 01345
노원구


127
주민등록번호 뒷 자리 7자리 중 첫째 자리는 성별을 나타내는데, 1, 3은 남자 2, 4는 여자를 의미한다. 사용자로부터 13자리의 주민등록번호를 입력 받은 후 성별 (남자, 여자)를 출력하는 프로그램을 작성하라.

>> 주민등록번호: 821010-1635210
남자

In [67]:
주민번호 =input("주민등록번호: ")
주민등록번호 = 주민번호[8]
if 주민등록번호== ("1" or "3"):
    print("남자")
else:
    print("여자")

주민등록번호: 980423-202020
여자


128
주민등록번호의 뒷 자리 7자리 중 두번째와 세번째는 지역코드를 의미한다.    
주민 등록 번호를 입력 받은 후 출생지가 서울인지 아닌지 판단하는 코드를 작성하라       

지역코드	출생지          
00 ~ 08	서울         
09 ~ 12	부산

In [69]:
주민번호 = input("주민등록번호: ")
출생지= 주민번호[9:11]
if int(출생지)<=8:
    print("서울")
else:
    print("부산")

주민등록번호: 980423-0091123
부산


129
주민등록번호는 13자리로 구성되는데 마지막 자리수는 주민등록번호의 유효성을 체크하는데 사용된다. 먼저 앞에서부터 12자리의 숫자에 2, 3, 4, 5, 6, 7, 8, 9, 2, 3, 4, 5를 차례로 곱한 뒤 그 값을 전부 더한다. 연산 결과 값을 11로 나누면 나머지가 나오는데 11에서 나머지를 뺀 값이 주민등록번호의 마지막 번호가 된다.

In [73]:
num = input("주민등록번호: ")
calc1 = int(num[0]) * 2 + int(num[1]) * 3 + int(num[2]) * 4 + int(num[3]) * 5 + int(num[4]) * 6 + \
        int(num[5]) * 7 + int(num[7]) * 8 + int(num[8]) * 9 + int(num[9]) * 2 + int(num[10])* 3 + \
        int(num[11])* 4 + int(num[12]) * 5
calc2 = 11 - (calc1%11)
calc3 = str(calc2)


if num[-1]== calc3[-1]:
    print("유효한 주민번호입니다.")
else:
    print("유효하지 않은 주민번호입니다")

주민등록번호: 821010-1635210
유효하지 않은 주민번호입니다


130
아래 코드는 비트코인의 가격 정보를 딕셔너리로 가져오는 코드이다.       

import requests      
btc = requests.get("https://api.bithumb.com/public/ticker/").json()['data']

In [74]:
import requests
btc = requests.get("https://api.bithumb.com/public/ticker/").json()['data']

In [78]:
change= float(btc['max_price'])-float(btc['min_price'])
open= float(btc['opening_price'])
close= float(btc['closing_price'])
max= float(btc['max_price'])

In [79]:
if (open+change) > max:
    print("상승장")
else:
    print("하락장")

상승장


131
for문의 실행결과를 예측하라.

과일 = ["사과", "귤", "수박"]
for 변수 in 과일:
    print(변수)

In [80]:
과일 = ["사과", "귤", "수박"]
for 변수 in 과일:
    print(변수)

사과
귤
수박


132
for문의 실행결과를 예측하라.

과일 = ["사과", "귤", "수박"]
for 변수 in 과일:
  print("#####")

In [81]:
과일 = ["사과", "귤", "수박"]
for 변수 in 과일:
  print("#####")

#####
#####
#####


133
다음 for 문과 동일한 기능을 수행하는 코드를 작성하세요.

for 변수 in ["A", "B", "C"]:
  print(변수)

In [83]:
for 변수 in ["A", "B", "C"]:
  print(변수)

A
B
C


In [82]:
print("A")
print("B")
print("C")

A
B
C


134
for문을 풀어서 동일한 동작을하는 코드를 작성하라.

for 변수 in ["A", "B", "C"]:
  print("출력:", 변수)

In [84]:
for 변수 in ["A", "B", "C"]:
  print("출력:", 변수)

출력: A
출력: B
출력: C


In [85]:
print("출력:", "A")
print("출력:", "B")
print("출력:", "C")

출력: A
출력: B
출력: C


135
for문을 풀어서 동일한 동작을 하는 코드를 작성하라.       

for 변수 in ["A", "B", "C"]:     
  b = 변수.lower()       
  print("변환:", b)      

In [88]:
for 변수 in ["A", "B", "C"]:
    b = 변수.lower()
    print("변환:", b)

변환: a
변환: b
변환: c


In [86]:
변수 = "A"
b = 변수.lower()
print("변환:", b)
변수 = "B"
b = 변수.lower()
print("변환:", b)
변수 = "C"
b = 변수.lower()
print("변환:", b)

변환: a
변환: b
변환: c


136  
다음 코드를 for문으로 작성하라.       

변수 = 10    
print(변수)   
변수 = 20    
print(변수)  
변수 = 30    
print(변수)     

In [89]:
for 변수 in [10, 20, 30]:
  print(변수)

10
20
30


137
다음 코드를 for문으로 작성하라.

print(10)
print(20)
print(30)

In [90]:
for 변수 in [10, 20, 30]:
  print(변수)

10
20
30


138
다음 코드를 for문으로 작성하라.

print(10)
print("-------")
print(20)
print("-------")
print(30)
print("-------")

In [91]:
for 변수 in [10, 20, 30]:
  print(변수)
  print("-------")

10
-------
20
-------
30
-------


139
다음 코드를 for문으로 작성하라.

print("++++")
print(10)
print(20)
print(30)

In [92]:
print("++++")
for 변수 in [10, 20, 30]:
  print(변수)

++++
10
20
30


140
다음 코드를 for문으로 작성하라.

print("-------")
print("-------")
print("-------")
print("-------")

In [94]:
for i in range(4):
    print("--------")

--------
--------
--------
--------


141
다음과 같이 판매가가 저장된 리스트가 있을 때 부가세가 포함된 가격을 for 문을 사용해서 화면에 출력하라. 단 부가세는 10원으로 가정한다.

리스트 = [100, 200, 300]       
110
210
310

In [1]:
list= [100,200,300]
for i in list:
    print(i+10)

110
210
310


142  
for 문을 사용해서 리스트에 저장된 값을 다음과 같이 출력하라.

리스트 = ["김밥", "라면", "튀김"]        
오늘의 메뉴: 김밥        
오늘의 메뉴: 라면       
오늘의 메뉴: 튀김       

In [3]:
menu = ["김밥", "라면", "튀김"]
for i in menu:
    print("오늘의 메뉴:"+i)

오늘의 메뉴:김밥
오늘의 메뉴:라면
오늘의 메뉴:튀김


143
리스트에 주식 종목이름이 저장돼 있다.

리스트 = ["SK하이닉스", "삼성전자", "LG전자"]       
저장된 문자열의 길이를 다음과 같이 출력하라.  

6    
4     
4      

In [4]:
stock = ["SK하이닉스","삼성전자","LG전자"]
for i in stock:
    print(len(i))

6
4
4


144
리스트에는 동물이름이 문자열로 저장돼 있다.       

리스트 = ['dog', 'cat', 'parrot']        
동물 이름과 글자수를 다음과 같이 출력하라.

dog 3
cat 3
parrot 6

In [5]:
animal = ['dog', 'cat', 'parrot']
for i in animal:
    print(i,len(i))

dog 3
cat 3
parrot 6


145
리스트에 동물 이름 저장돼 있다.

리스트 = ['dog', 'cat', 'parrot']       
for문을 사용해서 동물 이름의 첫 글자만 출력하라.

d
c
p

In [6]:
for i in animal:
    print(i[0])

d
c
p


146
리스트에는 세 개의 숫자가 바인딩돼 있다.

리스트 = [1, 2, 3]        
for문을 사용해서 다음과 같이 출력하라.       

3 x 1     
3 x 2          
3 x 3 

In [8]:
num=[1,2,3]
for i in num:
    print("3 x", i)

3 x 1
3 x 2
3 x 3


147
리스트에는 세 개의 숫자가 바인딩돼 있다.       
 
리스트 = [1, 2, 3]      
for문을 사용해서 다음과 같이 출력하라.        

3 x 1 = 3       
3 x 2 = 6       
3 x 3 = 9       

In [9]:
for i in num:
    print("3 x", i, "=", i*3)

3 x 1 = 3
3 x 2 = 6
3 x 3 = 9


148   
리스트에는 네 개의 문자열이 바인딩돼 있다.        
 
리스트 = ["가", "나", "다", "라"]      
for문을 사용해서 다음과 같이 출력하라.        
 
나    
다    
라  

In [11]:
kor= ["가", "나", "다", "라"]
for i in kor[1:]:
    print (i)

나
다
라


149
리스트에는 네 개의 문자열이 바인딩돼 있다.

리스트 = ["가", "나", "다", "라"]
for문을 사용해서 다음과 같이 출력하라.

가
다

In [12]:
for i in kor[::2]:
    print(i)

가
다


150
리스트에는 네 개의 문자열이 바인딩돼 있다.

리스트 = ["가", "나", "다", "라"]
for문을 사용해서 다음과 같이 출력하라.

라
다
나
가

In [13]:
for i in kor[::-1]:
    print(i)

라
다
나
가
